### PR TITLE
Add Web API integration tests for GetChargeLinksByMeteringPointId endpoint.

### PR DESCRIPTION
**Pull request by prtandrup**
## Description

This PR adds web api integration tests targeting the GetChargeLinksByMeteringPointId endpoint.

## References

* #792

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        private readonly WebApplicationFactory<Charges.WebApi.Startup> _factory;
+
+        public ChargeLinksControllerTests(WebApplicationFactory<Charges.WebApi.Startup> factory)
+        {
+            _factory = factory;
+        }
+
+        [Theory]
+        [InlineData("/ChargeLinks/GetChargeLinksByMeteringPointId?meteringPointId=1")]
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:38]: I would prefer to have the base url "/ChargeLinks/GetChargeLinksByMeteringPointId?meteringPointId=" as a `const`, and only `meteringPointId` as a paramter to each test. Url could then be concatenated:
`var response = await client.GetAsync($"{baseUrl}{meteringPointId}");`

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            var response = await client.GetAsync(url);
+            var jsonString = await response.Content.ReadAsStringAsync();
+            var result = JsonSerializer.Deserialize<List<ChargeLinkDto>>(jsonString, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+
+            // Assert
+            Assert.Equal("40000", result![0].ChargeId);
+        }
+
+        [Theory]
+        [InlineData("/ChargeLinks/GetChargeLinksByMeteringPointId?meteringPointId=404")]
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:70]: how about using a negative integer for the Id? That way this won't break if we ever get a metering point with id 404 :)

Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:70]: Nice unknown meteringPointId number :-)

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        [Theory]
+        [InlineData("/ChargeLinks/GetChargeLinksByMeteringPointId?meteringPointId=1")]
+        public async Task GetChargeLinksByMeteringPointId_WhenIdHasChargeLinks_ReturnsOkAndCorrectContentType(string url)
+        {
+            // Arrange
+            var client = _factory.CreateClient();
+
+            // Act
+            var response = await client.GetAsync(url);
+            var contentType = response.Content.Headers.ContentType!.ToString();
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:46]: move this line to assert

**Review by HenrikSommer — APPROVED** [review]
Approved with a suggestion.

**Review by lasrinnil — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs (hunk 2 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:70]: Right now the backend returns charge links on any int input, except 404. Maybe something to do later?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs (hunk 1 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs]: Thx, fixed with 4a44258f

**Pull request merged**